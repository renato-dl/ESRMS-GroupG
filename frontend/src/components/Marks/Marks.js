import React from 'react';
//import { api } from '../../services/api';
import "./Marks.scss"

import _ from 'lodash';
import {Icon, Table, Header, Grid, Image} from 'semantic-ui-react';


const columns = _.times(16, (i) => (
  <Grid.Column key={i}>
    <Image src='https://react.semantic-ui.com/images/wireframe/image.png' />
  </Grid.Column>
))


export class Marks extends React.Component{
    constructor(props)
    {
      super(props)
      this.state={
      Marks:[
        {subject:"subj1",marks:"100"},
        {subject:"subj2",marks:"90"},
        {subject:"subj3",marks:"80"}
      ]
      }
    }
    // async componentDidMount(){
    //     const responce=await api.marks.getMarks(1);
    //     if(responce){
    //         this.setState({Marks:responce.data})
    //     }
    // }

    // selectMarks=async(studentID)=>{
    //     console.log(studentID);
    //     this.props.history.push('/marks')
    // }
      render(){
        return (
          <>
            <h3 className="contentHeader"> 
                <Icon name='sort numeric up' /> Grades of Nume Surname 2019-2020
              </h3>
            <div className="Marks-container">
              <div className="tableContainer">
              <Table basic='very' celled collapsing>
                <Table.Header>
                  <Table.Row>
                  <Table.HeaderCell><Icon name='book' />SUBJECT</Table.HeaderCell>
                  <Table.HeaderCell><Icon name='sort numeric up' />GRADE</Table.HeaderCell>
                  </Table.Row>
                </Table.Header>

                <Table.Body>
                  {this.state.Marks.map((mark, index)=>
                  
                  <Table.Row  key = {index}>
                    <Table.Cell >
                      <Header as='h4' image>
                       {/* <Icon name='angle right' size=" small"/> */}
                        <Header.Content>
                          {mark.subject}
                          {/* <Header.Subheader>Human Resources</Header.Subheader> */}
                        </Header.Content>
                      </Header>
                    </Table.Cell>
                    <Table.Cell positive>
                      {mark.marks}
                      </Table.Cell>
                  </Table.Row>

                  )}

                </Table.Body>
              </Table>
              </div>

{/* 
            <h2 className="title">Student{this.props.match.params.studentID}'s score:</h2>
            {this.state.Marks.map((mark, index)=>
            <p key={index}>
              subject:{mark.subject},
              marks:{mark.marks}
            </p>
              )}
 */}

            </div>



            
            <Grid>{columns}</Grid>



          </>
        )
    }
    }