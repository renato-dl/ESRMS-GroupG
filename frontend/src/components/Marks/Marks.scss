.Marks-container{
    width:calc(100% - 260px);
    padding: 0 30px 30px 30px;
}
/* .tableContainer{
    background-color:#fff;
    padding:15px;
    width:auto;
} */
.ui.basic.table{
    text-align: center;
    padding:10px;
    background-color:#FFF6F6;
    border: 15px solid #FFF6F6;
}