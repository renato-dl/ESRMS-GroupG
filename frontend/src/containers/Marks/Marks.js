import React from 'react';
import { api } from '../../services/api';
import './Marks.scss';

import { Header, Icon} from 'semantic-ui-react';

export class Marks extends React.Component{
    constructor(props)
    {
      super(props)
      this.state={
      Marks:[
        {subject:"subj1",marks:"100"},
        {subject:"subj2",marks:"90"},
        {subject:"subj3",marks:"80"}
      ]
      }
    }
    // async componentDidMount(){
    //     const responce=await api.marks.getMarks(1);
    //     if(responce){
    //         this.setState({Marks:responce.data})
    //     }
    // }

    // selectMarks=async(studentID)=>{
    //     console.log(studentID);
    //     this.props.history.push('/marks')
    // }
      render(){
        return (
            <div className="Marks-container">
              <h3 className="contentHeader"> 
        <Icon name='sort numeric up' /> Grades of Nume Surname 2019-2020</h3>
            <h2 className="title">Student{this.props.match.params.studentID}'s score:</h2>
            {this.state.Marks.map(mark=>
            <p>
              subject:{mark.subject},
              marks:{mark.marks}
            </p>
              )}
            </div>
        )
    }
    }