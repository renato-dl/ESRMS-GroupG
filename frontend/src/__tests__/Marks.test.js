import React from 'react';
import {Marks} from '../components/Marks/Marks';
import {render} from '@testing-library/react'
import {getRouterPropsForTest} from '../utils';

describe('Testing Marks component', () => {
  
  test('Test if component is rendered', () => {
    const component = render(
      <Marks {...getRouterPropsForTest()} />
    );
  });

});
