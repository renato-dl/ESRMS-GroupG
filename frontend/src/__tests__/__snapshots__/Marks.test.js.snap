// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Marks component Testing snapshot 1`] = `
<div
  className="Marks-container"
>
  <h2
    className="title"
  >
    Student
    's score:
  </h2>
</div>
`;
